### PR TITLE
untracked inputs in constraints

### PR DESCRIPTION
**Pull request by avikchaudhuri**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #109037

Differential Revision: [D49157009](https://our.internmc.facebook.com/intern/diff/D49157009/)

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @chenyang78 @aakhundov @kadeng

### CODE REVIEW

File: torch/_export/exported_program.py [hunk 1]
```diff
@@ -337,7 +337,7 @@ def _process_constraints(
         val = placeholder_nodes[input_dim.input_name].meta["val"]
         assert isinstance(val, FakeTensor)
         symint = val.shape[input_dim.dim]
-        assert isinstance(symint, SymInt)
+        assert isinstance(symint, SymInt), f"Expected SymInt but got {symint}: {type(symint)}"
```
Comment (zhxchen17) [inline torch/_export/exported_program.py:340]: @avikchaudhuri does this mean we still ran into assertion failure?

Comment (avikchaudhuri) [reply torch/_export/exported_program.py:340]: Only if you try passing a 0 or 1 dimensional tensor in KJT, yes. In all other cases this assert should not fail.

Comment (avikchaudhuri) [reply torch/_export/exported_program.py:340]: For KJT you get an int that is 0 or 1.

Comment (zhxchen17) [reply torch/_export/exported_program.py:340]: ah I see, it makes sense now.

**Review by zhxchen17 — APPROVED** [review]

**Pull request closed without merging**